pkg/updater/service/datadog_agent.go

```diff
@@ -120,3 +139,82 @@ func StartAgentExperiment() error {
 func StopAgentExperiment() error {
 	return startUnit(agentUnit)
 }
+
+func setInstallInfo() {
+	// avoid rewriting the files if they already exist
+	if _, err := os.Stat(installInfoFile); err == nil {
+		log.Info("Install info file already exists, skipping")
+		return
+	}
+	tool, version := getToolVersion()
+	if err := writeInstallInfo(tool, version); err != nil {
+		log.Warnf("Failed to write install info: %s", err)
+	}
+	if err := writeInstallSignature(); err != nil {
+		log.Warnf("Failed to write install signature: %s", err)
+	}
+}
+
+func cleanInstallInfo() {
+	if err := os.Remove(installInfoFile); err != nil && !os.IsNotExist(err) {
+		log.Warnf("Failed to remove install info file: %s", err)
+	}
+	if err := os.Remove(installSigFile); err != nil && !os.IsNotExist(err) {
+		log.Warnf("Failed to remove install signature file: %s", err)
+	}
+}
```

Comment:
Those should be part of the `installinfo` package as generic helpers so other can reuse them when needed.

Same with most new method in this file

Reply:
Do you want the same failure behaviours for setInstallInfo and cleanInstallInfo, if so do I just move them to installinfo with the matching tests?

Reply:
I think we could return an error to keep them as pure helper and let the caller log it if needed

Reply:
in this case we want to pass through each step even if one fails